api/src/main/java/jakarta/config/Configuration.java

```diff
@@ -25,13 +25,47 @@
 import java.lang.annotation.Target;
 
 /**
- * <p>
- *     The annotation to be used to mark the interface to be handled by the injection support.
- * </p>
+ * <p>Indicates that the annotated class is a <em>configuration interface class</em> to be handled by the Jakarta Config
+ * implementation injection support.</p>
+ *
+ * <p>The <em>configuration interface class</em> is resolved with portion of application's
+ * <em>persistent configuration</em> identified by <em>configuration path</em>.
+ *
+ * <p>The terms <em>configuration interface class</em>, <em>configuration key</em>, <em>configuration path</em>, <em>canonical
+ * representation</em>, <em>persistent configuration</em>, <em>resolve</em>, and others are used here as defined in
+ * the Jakarta Config specification.</p>
+
+ * <p><strong>\u26A0 Caution:</strong> you are reading an incomplete draft specification that is subject to change.</p>
  */
 @Target({ElementType.TYPE})
 @Retention(RetentionPolicy.RUNTIME)
 @Documented
 public @interface Configuration {
 
+    /**
+     * The <em>configuration path</em> may identify where, in a given application's <em>persistent configuration</em>,
+     * the configuration relevant for the annotated configuration class may be found.
+     *
+     * <p>The configuration path is a <em>canonical representation</em> of a <em>configuration key</em>,
+     * as defined in the Jakarta Config specification.</p>
+     *
+     * <p>The configuration path represented by this element is effectively a hint to a Jakarta Config implementation
+     * about where, in an application's persistent configuration, a portion of that configuration relevant to the
+     * annotated configuration interface class may be found.</p>
+     *
+     * @return a {@link String} representation of a configuration path.
+     */
+    String path() default "";
+
+
+    /**
+     * The <em>separator</em> used in the <em>configuration path</em> to identify the relevant <em>configuration key</em>
+     * in application's <em>persistent configuration</em>.
+     *
+     * <p>The default separator is <em>.</em> (dot) symbol.</p>
+     *
+     * @return a {@link String} representation of a configuration path separator.
+     */
+    String separator() default ".";
```

Comment:
In the first release, we might even not to add this separator. MP just uses dot without an alternative separator. I have not got any request for supporting other separators.

Reply:
@Emily-Jiang Do you want to remove it or are you fine with being it a possibility and just saying it is not a must at the moment?